spectre-meltdown-checker.sh

```diff
@@ -1217,10 +1198,23 @@ check_variant3()
 	elif [ -z "$msg" ]; then
```

Comment:
Could you try with a Xen system under the 4.14.14 series ? Or any other recent version that has the `/sys/devices/system/cpu/vulnerabilities` hierarchy? In that case the "$msg" should be filled, and I'm curious whether the kernel would flag the system as vulnerable or not for variant3! We might want to override what it says in that case

Reply:
So, with Linux 4.14.14 and Xen 4.8.3-pre:
```
root@makeba:~# cat /sys/devices/system/cpu/vulnerabilities/spectre_v1
Vulnerable
root@makeba:~# cat /sys/devices/system/cpu/vulnerabilities/spectre_v2
Vulnerable: Minimal generic ASM retpoline
root@makeba:~# cat /sys/devices/system/cpu/vulnerabilities/meltdown 
Vulnerable
```

Reply:
Interestingly, as seen in https://github.com/speed47/spectre-meltdown-checker/issues/59, the machine is advertised as meltdown-vulnerable, but:
`[    0.000000] Kernel/User page tables isolation: disabled on XEN PV.`
due to arch/x86/mm/pti.c (was kaiser.c previously):
```
        if (hypervisor_is_type(X86_HYPER_XEN_PV)) {
                pti_print_if_insecure("disabled on XEN PV.");
                return;
        }
```

Reply:
That's what I expected, so the Xen check should be done even if not [ -z "$msg" ], i.e. regardless of the fact that the `/sys/devices/system/cpu/vulnerabilities` hierarchy is there or not: the kernel is wrong and we are right in that case !